### PR TITLE
45 coding bugs in vrdr java library in deathrecordvoidmessagejava and deathrecordupdatemessagejava

### PR DESCRIPTION
**Pull request by consultantusa**

### CODE REVIEW

File: src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java [hunk 1]
```diff
@@ -72,6 +73,10 @@ public void setBlockCount(Integer value) {
         }
         else {
             messageParameters.setParameter("block_count", (IntegerType)null);
+            messageParameters.setParameter("block_count", (UnsignedIntType)null);
+            if (value != null && value > 0) {
+                messageParameters.addParameter("block_count", new IntegerType(value));
+            }
```
Comment (johnarleyburns) [inline src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java:79]: Should this be new UnsignedIntType ?

File: src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java [hunk 2]
```diff
@@ -72,6 +73,10 @@ public void setBlockCount(Integer value) {
         }
         else {
             messageParameters.setParameter("block_count", (IntegerType)null);
+            messageParameters.setParameter("block_count", (UnsignedIntType)null);
```
Comment (johnarleyburns) [inline src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java:76]: Should the line above be removed?

File: src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java [hunk 3]
```diff
@@ -72,6 +73,10 @@ public void setBlockCount(Integer value) {
...
+            if (value != null && value > 0) {
```
Comment (johnarleyburns) [inline src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java:77]: We should add a test case for this.

**Review by johnarleyburns — CHANGES_REQUESTED** [review]

File: src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java (hunk 1 above)
Comment (consultantusa) [reply src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java:77]: Yes. fixed. Thank you.

File: src/test/java/edu/gatech/VRDR/MessagingTest.java [hunk 4]
```diff
@@ -637,7 +639,20 @@ public void testCreateAckForDeathRecordVoidMessage() {
...
+	voidMessage.setBlockCount(value);
+	assertEquals(voidMessage.getBlockCount(), null);
+	value = 2;
+	voidMessage.setBlockCount(value);
+	assertEquals(voidMessage.getBlockCount(), 2); 
+    }
+
+//import org.hl7.fhir.r4.model.IntegerType;
+//import org.hl7.fhir.r4.model.UnsignedIntType;
+
```
Comment (johnarleyburns) [inline src/test/java/edu/gatech/VRDR/MessagingTest.java:656]: Please delete these comments, also do a test with block count 0 or negative to make sure it doesn’t set the value

File: src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java (hunk 1 above)
Comment (johnarleyburns) [reply src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java:78]: It still says “new IntegerType”

**Review by johnarleyburns — CHANGES_REQUESTED** [review]

File: src/test/java/edu/gatech/VRDR/MessagingTest.java [hunk 5]
```diff
@@ -637,6 +639,19 @@ public void testCreateAckForDeathRecordVoidMessage() {
         assertEquals(voidMessage.getNCHSIdentifier(), ack.getNCHSIdentifier());
         assertEquals(voidMessage.getBlockCount(), ack.getBlockCount());
     }
+	
+    public void testSetBlockCountDeathRecordVoidMessage() {
+	DeathRecordVoidMessage voidMessage = new DeathRecordVoidMessage();
+// 	Integer value = null;
```
Comment (johnarleyburns) [inline src/test/java/edu/gatech/VRDR/MessagingTest.java:645]: Please remove commented code

File: src/test/java/edu/gatech/VRDR/MessagingTest.java [hunk 6]
```diff
@@ -637,6 +639,19 @@ public void testCreateAckForDeathRecordVoidMessage() {
...
+// 	voidMessage.setBlockCount(value);
+// 	assertEquals(voidMessage.getBlockCount(), null);
+	Integer value = 2;
+	voidMessage.setBlockCount(value);
+	System.out.println("******value******="+value);
```
Comment (johnarleyburns) [inline src/test/java/edu/gatech/VRDR/MessagingTest.java:650]: Please don’t add system out println to the tests, only assertions.

**Review by johnarleyburns — CHANGES_REQUESTED** [review]

File: src/test/java/edu/gatech/VRDR/MessagingTest.java (hunk 6 above)
Comment (consultantusa) [reply src/test/java/edu/gatech/VRDR/MessagingTest.java:650]: Mr. Burns, I am still working on the issue with debugging frags. I will notify you again once I am done. Thank you.

**Review by johnarleyburns — CHANGES_REQUESTED** [review]
Close to complete, please squash all commits into one

**Review by johnarleyburns — APPROVED** [review]

File: src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java [hunk 7]
```diff
@@ -67,11 +68,8 @@ public Integer getBlockCount() {
     }
 
     public void setBlockCount(Integer value) {
```
[the remark this answers is not in the document]
Comment (whichwit) [reply src/main/java/edu/gatech/chai/VRDR/messaging/DeathRecordVoidMessage.java:70]: I apologize @consultantusa, the review was never submitted even though I had written down the comment. Please review and let's discuss.
